src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -56,18 +56,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String username = request.getParameter(USERNAME_PARAMETER_KEY);
     String bio = request.getParameter(BIO_PARAMETER_KEY);
     boolean edit = Boolean.parseBoolean(request.getParameter(EDIT_PARAMETER_KEY));
+    Optional<UserProfile> userProfile = profileData.queryProfileByUsername(username);
+    boolean isUserNameTaken = userProfile.isPresent() && !email.equals(userProfile.get().getEmail());
```

Comment:
Create this boolean right before it is used in the next if statement.

Reply:
Can you explain why and what you mean?

Reply:
Normally you prefer to initialize a variable right before it is used, in this case the boolean ```isUserNameTaken``` is not needed in the current if statement so you can initialize it right after.

Reply:
I agree with Bryan.

For example, if `areValidParamters()` on line 62 returns false, then you don't even have to initialize the variables `userProfile` and `isUserNameTaken`. That saves some time because you can skip the query to the database (i.e. the call to `queryProfileByUsername()`).

So ideally, you should move lines 59-60 to line 66.